File: android/worker/app/src/main/java/com/workforcex/worker/api/WorkforceXApi.java
package com.workforcex.worker.api;

import java.util.List;
import okhttp3.MultipartBody;
import okhttp3.RequestBody; // Added import
import retrofit2.Call;
import retrofit2.http.*;

public interface WorkforceXApi {

    @POST("api/auth/register")
    Call<RegisterResponse> register(@Body RegisterRequest request);

    @POST("api/auth/login")
    Call<LoginResponse> login(@Body LoginRequest request);

    @PUT("api/worker/profile")
    Call<WorkerProfileResponse> saveWorkerProfile(
            @Header("Authorization") String token,
            @Body WorkerProfileRequest request);

    @GET("api/worker/profile")
    Call<WorkerProfileResponse> getWorkerProfile(@Header("Authorization") String token);

    @GET("api/jobs/browse")
    Call<List<JobBrowseItem>> browseJobs(@Header("Authorization") String token);

    @Multipart
    @POST("api/worker/resume")
    Call<ResumeParseResult> uploadResume(
            @Header("Authorization") String token,
            @Part MultipartBody.Part file);

    // Applications
    @POST("api/applications/{jobId}")
    Call<JobApplication> applyToJob(
            @Header("Authorization") String token,
            @Path("jobId") String jobId);

    @GET("api/applications/my")
    Call<List<JobApplication>> getMyApplications(@Header("Authorization") String token);

    @GET("api/notifications")
    Call<List<Notification>> getUnreadNotifications(@Header("Authorization") String token);

    // Verification
    @GET("api/verification/status")
    Call<List<Verification>> getVerificationStatus(@Header("Authorization") String token);

    @Multipart
    @POST("api/verification/upload")
    Call<Document> uploadDocument(
            @Header("Authorization") String token,
            @Part("type") RequestBody type,
            @Part MultipartBody.Part file);
}

File: android/worker/shared/src/commonMain/kotlin/com/workforcex/shared/WorkforceXApi.kt
package com.workforcex.shared

import com.workforcex.shared.models.*
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Call
import retrofit2.http.*

interface WorkforceXApi {

    @POST("api/auth/register")
    fun register(@Body request: RegisterRequest): Call<RegisterResponse>

    @POST("api/auth/login")
    fun login(@Body request: LoginRequest): Call<LoginResponse>

    @PUT("api/worker/profile")
    fun saveWorkerProfile(
        @Header("Authorization") token: String,
        @Body request: WorkerProfileRequest
    ): Call<WorkerProfileResponse>

    @GET("api/worker/profile")
    fun getWorkerProfile(@Header("Authorization") token: String): Call<WorkerProfileResponse>

    @GET("api/jobs/browse")
    fun browseJobs(@Header("Authorization") token: String): Call<List<JobBrowseItem>>

    @Multipart
    @POST("api/worker/resume")
    fun uploadResume(
        @Header("Authorization") token: String,
        @Part file: MultipartBody.Part
    ): Call<ResumeParseResult>

    // Applications
    @POST("api/applications/{jobId}")
    fun applyToJob(
        @Header("Authorization") token: String,
        @Path("jobId") jobId: String
    ): Call<JobApplication>

    @GET("api/applications/my")
    fun getMyApplications(@Header("Authorization") token: String): Call<List<JobApplication>>

    @GET("api/notifications")
    fun getUnreadNotifications(@Header("Authorization") token: String): Call<List<Notification>>

    // Verification
    @GET("api/verification/status")
    fun getVerificationStatus(@Header("Authorization") token: String): Call<List<Verification>>

    @Multipart
    @POST("api/verification/upload")
    fun uploadDocument(
        @Header("Authorization") token: String,
        @Part("type") type: RequestBody,
        @Part file: MultipartBody.Part
    ): Call<Document>

    // Skills
    @GET("api/skills")
    fun getSkills(): Call<List<Skill>>
}